morph.go

```diff
@@ -70,11 +71,13 @@ func SetSatementTimeoutInSeconds(n int) EngineOption {
 	}
 }
 
-// WithLockKey creates a lock table in the database so that the migrations are
-// guaranteed to be executed from a single instance.
-func WithLockKey(key string) EngineOption {
+// WithLock creates a lock table in the database so that the migrations are
+// guaranteed to be executed from a single instance. The key is used for naming
+// the mutex.
+func WithLock(ctx context.Context, key string) EngineOption {
 	return func(m *Morph) {
 		m.config.LockKey = key
+		m.config.LockContext = ctx
```

Comment:
context as a struct field isn't really [idiomatic](https://pkg.go.dev/context#section-documentation).

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.

Can we pass this in `New`  as a separate param?

Reply:
Sure, the params in `New` was quite bloated so didn't want to add there in the first place. But it is what it is. Fun fact I didn't exactly know that context as a field was not right, but it felt wrong to add it as a field TBH.